### PR TITLE
Write out submitty details to conf folder

### LINKED ISSUE #1750: Split apart /usr/local/submitty/.setup/submitty_conf.json and move to /usr/local/submitty/conf
This file looks like:
```
{
  "submitty_install_dir": "/usr/local/submitty",
  "submitty_repository": "/usr/local/submitty/GIT_CHECKOUT_Submitty",
  "submitty_tutorial_dir": "/usr/local/submitty/GIT_CHECKOUT_Tutorial",
  "submitty_data_dir": "/var/local/submitty",
  "hwphp_user": "hwphp",
  "hwcgi_user": "hwcgi",
  "hwcron_user": "hwcron",
  "hwcronphp_group": "hwcronphp",
  "course_builders_group": "course_builders",
  "num_untrusted": 60,
  "first_untrusted_uid": 900,
  "first_untrusted_gid": 900,
  "hwcron_uid": 1003,
  "hwcron_gid": 1006,
  "hwphp_uid": 1001,
  "hwphp_gid": 1004,
  "database_host": "/var/run/postgresql",
  "database_user": "hsdbu",
  "database_password": "hsdbu",
  "authentication_method": "PamAuthentication",
  "submission_url": "http://192.168.56.101",
  "tagrading_url": "http://192.168.56.101/hwgrading",
  "cgi_url": "http://192.168.56.101/cgi-bin",
  "autograding_log_path": "/var/local/submitty/logs/autograding",
  "site_log_path": "/var/local/submitty/logs",
  "num_grading_scheduler_workers": 5,
  "debugging_enabled": true,
  "institution_name": "",
  "username_change_text": "Submitty welcomes individuals of all ages, backgrounds, citizenships, disabilities, sex, education, ethnicities, family statuses, genders, gender identities, geographical locations, languages, military experience, political views, races, religions, sexual orientations, socioeconomic statuses, and work experiences. In an effort to create an inclusive environment, you may specify a preferred name to be used instead of what was provided on the registration roster.",
  "institution_homepage": ""
}
```

We should split apart this file such that the privileged information (like the DB stuff) is in its own file with appropriate permissions and the non-privileged information (most everything else) is in another file with higher permissions.

We should then have these files being written to `/usr/local/submitty/conf`.

### PR DESCRIPTION
**Pull request by MasterOdin**
Closes #1750 (though leaves the files).

Here's the permissions of these new files:
```
root@vagrant:/usr/local/submitty# ls -ld /usr/local/submitty/conf
drwxr-x--x 2 root course_builders 4096 Feb 21 23:37 /usr/local/submitty/conf
root@vagrant:/usr/local/submitty# ls -la /usr/local/submitty/conf
total 20
drwxr-x--x  2 root course_builders 4096 Feb 21 23:37 .
drwxr-x--x 15 root course_builders 4096 Feb 21 23:37 ..
-rw-r-----  1 root root             169 Feb 21 23:37 autograding_workers.json
-r--r-----  1 root hwcron           133 Feb 21 23:37 database_conf.json
-r--r-----  1 root course_builders 1460 Feb 21 23:37 submitty_conf.json
```

I also moved autograding_workers.json into this new folder as it's not currently hooked up to anything and I'd like that file to just be in the proper place to start with.

Made also improvement on checking for existance and read/write of the install/data directories, as well as make it so that we can change these directories by passing in different runtime arguments to CONFIGURE_SUBMITTY.py.

### CODE REVIEW

**Comment by MasterOdin** [comment]
@bmcutler 

Here's my current proposal for the following conf files and their permissions. I've split apart the information into different files/sections and then restricted it as much as possible to the relevant users that need to read them.

```
-r--r-----  1 root   hwcron           169 Mar 21 15:55 autograding_workers.json
-r--r-----  1 hwphp www-data         181 Mar 21 15:55 database.json
-r--r--r--  1 root   root             518 Mar 21 15:55 paths.json
-r--r--r--  1 root   root             553 Mar 21 15:55 submitty.json
-r--r-----  1 root   hwcron           366 Mar 21 15:55 users.json
```

Where the contents for each file are:
```
root@vagrant:/usr/local/submitty/conf# cat autograding_workers.json
{
    "primary": {
        "address": "",
        "capabilities": [
            "default"
        ],
        "num_autograding_workers": 5,
        "username": ""
    }
}
```

```
root@vagrant:/usr/local/submitty/conf# cat database.json
{
  "authentication_method": "PamAuthentication",
  "database_host": "/var/run/postgresql",
  "database_user": "hsdbu",
  "database_password": "hsdbu",
  "debugging_enabled": true
}
```
```
root@vagrant:/usr/local/submitty/conf# cat paths.json
{
  "submitty_install_dir": "/usr/local/submitty",
  "submitty_repository": "/usr/local/submitty/GIT_CHECKOUT_Submitty",
  "submitty_tutorial_dir": "/usr/local/submitty/GIT_CHECKOUT_Tutorial",
  "submitty_data_dir": "/var/local/submitty",
  "autograding_log_path": "/var/local/submitty/logs/autograding",
  "site_log_path": "/var/local/submitty/logs",
  "submission_url": "http://192.168.56.101",
  "vcs_url": "/git",
  "tagrading_url": "http://192.168.56.101/hwgrading",
  "cgi_url": "http://192.168.56.101/cgi-bin"
}
```
```
root@vagrant:/usr/local/submitty/conf# cat submitty.json
{
  "institution_name": "",
  "username_change_text": "Submitty welcomes individuals of all ages, backgrounds, citizenships, disabilities, sex, education, ethnicities, family statuses, genders, gender identities, geographical locations, languages, military experience, political views, races, religions, sexual orientations, socioeconomic statuses, and work experiences. In an effort to create an inclusive environment, you may specify a preferred name to be used instead of what was provided on the registration roster.",
  "institution_homepage": ""
}
```
```
root@vagrant:/usr/local/submitty/conf# cat users.json
{
  "num_grading_scheduler_workers": 5,
  "num_untrusted": 60,
  "first_untrusted_uid": 900,
  "first_untrusted_gid": 900,
  "hwcron_uid": 1003,
  "hwcron_gid": 1006,
  "hwphp_uid": 1001,
  "hwphp_gid": 1004,
  "hwphp_user": "hwphp",
  "hwcgi_user": "hwcgi",
  "hwcron_user": "hwcron",
  "hwcronphp_group": "hwcronphp",
  "course_builders_group": "course_builders"
}
```

This should be that hwcron/root can access all files and hwphp/www-data can access the DB information and paths (can get rid of the master.ini file) and all users should be able to get the basic path information (where is submitty installed, where is the data directory, etc.) and just misc other info (right now basically just institution name, url, and message to show to users on changing their name) as this information is trivial to figure out via submissions if you really cared to get it and it wasn't the defaults.

**Comment by bmcutler** [comment]
Silly question, but why are we naming the directory "conf"?  Can we name it "config"?
(let's use a single abbreviation in our system unless there is a good reason not to)

**Comment by bmcutler** [comment]
The file permissions don't match what you've proposed above...

root@vagrant:/usr/local/submitty/conf# ls -l
total 16
-r--r----- 1 hwcron hwcron  178 Mar 29 17:52 autograding_workers.json
-r--r----- 1 hwcron hwcron  182 Mar 29 17:52 database.json
-r--r--r-- 1 hwcron hwcron 1134 Mar 29 17:52 submitty.json
-rw-r----- 1 hwcron hwcron  366 Mar 29 17:52 users.json

File: bin/submitty_autograding_shipper.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def initialize(untrusted_queue):
 
 # ==================================================================================
 def update_all_foreign_autograding_workers():
-    all_workers_json = os.path.join(SUBMITTY_INSTALL_DIR, "conf", "autograding_workers.json")
+    all_workers_json = os.path.join(SUBMITTY_INSTALL_DIR, 'conf', "autograding_workers.json")
```
Comment (bmcutler) [inline bin/submitty_autograding_shipper.py]: should this be config?

Comment (MasterOdin) [reply bin/submitty_autograding_shipper.py]: Yeah, my bad. Fixed.

**Pull request merged**